src/SonarScanner.MSBuild.Tasks/Targets/SonarQube.Integration.targets

```diff
@@ -397,12 +397,15 @@
       <ErrorLog>$(RazorCompilationErrorLog)</ErrorLog>
       <!-- Keep the path so that we could set SonarQubeSetting with sonar.language.roslyn.reportFilePaths -->
       <RazorSonarErrorLog>$(RazorCompilationErrorLog)</RazorSonarErrorLog>
-    </PropertyGroup>
-    <PropertyGroup>
       <SonarTemporaryProjectSpecificOutDir>$(ProjectSpecificOutDir).tmp</SonarTemporaryProjectSpecificOutDir>
     </PropertyGroup>
-    <MoveDirectory SourceDirectory="$(ProjectSpecificOutDir)" DestinationDirectory="$(SonarTemporaryProjectSpecificOutDir)" />
-    <MakeDir Directories="$(ProjectSpecificOutDir)" />
+    <ItemGroup>
+      <CoreCompileOutFiles Include="$(ProjectSpecificOutDir)\**\*.*"/>
+    </ItemGroup>
+    <Move
+      SourceFiles="@(CoreCompileOutFiles)"
+      DestinationFolder="$(SonarTemporaryProjectSpecificOutDir)\%(RecursiveDir)" />
+    <Message Importance="high" Text="Sonar: Preparing for Razor compilation, moved files ($(CoreCompileOutFiles)) to $(SonarTemporaryProjectSpecificOutDir)." />
```

Comment:
Out of curiosity, what is the format of the log, once it is printed? Are the SourceFiles coma delimited?

Reply:
>Sonar: Preparing for Razor compilation, moved files (C:\workspace\sonar-scanner-msbuild\TestResults\Deploy_Andrei 2022-02-04 10_03_42\3\Outputs\0\foo\bar.txt;C:\workspace\sonar-scanner-msbuild\TestResults\Deploy_Andrei 2022-02-04 10_03_42\3\Outputs\0\ProjectInfo.xml) to C:\workspace\sonar-scanner-msbuild\TestResults\Deploy_Andrei 2022-02-04 10_03_42\3\Outputs\0.tmp.